### PR TITLE
Fix step slug styling when user uses custom names

### PR DESCRIPTION
**Pull request by ravikiranvm**
Fixes OPS-1928.

## Additional Notes

- Updated the CSS for the step name hover label.
- Replaced the previous negative right offset with a positive [left: calc(100% + 10px)]
- This ensures the label always appears to the right of the node, regardless of the step name length, and prevents overlap with the node.

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

If there are UI/UX changes, include at least one of the following:

- Screenshots
Screenshot 2025-07-14 at 4 49 47 PM

- Loom video
- Preview deployment link

### CODE REVIEW

File: packages/react-ui/src/app/features/builder/flow-canvas/nodes/step-node.tsx [hunk 1]
```diff
@@ -153,10 +153,10 @@ const WorkflowStepNode = React.memo(
           {...{ [`data-${STEP_CONTEXT_MENU_ATTRIBUTE}`]: data.step!.name }}
         >
           <div
-            className="absolute text-accent-foreground text-sm opacity-0 transition-all duration-300 group-hover:opacity-100 "
+            className="absolute text-accent-foreground text-sm opacity-0 transition-all duration-300 group-hover:opacity-100"
             style={{
               top: `${OPS_NODE_SIZE.stepNode.height / 2 - 12}px`,
-              right: `-${OPS_NODE_SIZE.stepNode.width / 5}px`,
+              left: `calc(100% + 10px)`,
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/react-ui/src/app/features/builder/flow-canvas/nodes/step-node.tsx]: I don't think this suggestion is necessary at this stage.

File: packages/react-ui/src/app/features/builder/flow-canvas/nodes/step-node.tsx [hunk 2]
```diff
@@ -153,10 +153,10 @@ const WorkflowStepNode = React.memo(
           {...{ [`data-${STEP_CONTEXT_MENU_ATTRIBUTE}`]: data.step!.name }}
         >
           <div
-            className="absolute text-accent-foreground text-sm opacity-0 transition-all duration-300 group-hover:opacity-100 "
+            className="absolute text-accent-foreground text-sm opacity-0 transition-all duration-300 group-hover:opacity-100"
             style={{
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/react-ui/src/app/features/builder/flow-canvas/nodes/step-node.tsx:158]: The current implementation is already in consistent with the existing pattern in the current codebase. So I'm ignoring this suggestion for now.

File: packages/react-ui/src/app/features/builder/flow-canvas/nodes/step-node.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (cezudas) [reply packages/react-ui/src/app/features/builder/flow-canvas/nodes/step-node.tsx]: @ravikiranvm Hope you don't mind, this suggestion is generally good. `10` in `10px` is a "magic" number.

[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/react-ui/src/app/features/builder/flow-canvas/nodes/step-node.tsx]: @cezudas Done :)

**Review by cezudas — CHANGES_REQUESTED** [review]
There is one edge-case that needs to be addressed as well, use this WF as a reference:
https://dev.internal.openops.com/flows/2vSyLTOAERT4BUBpNoTy6?viewOnly=true

https://www.loom.com/share/421cad178ae24bd0acb3ae8d5178122f
Your PR branch:
Screenshot 2025-07-14 at 17 21 47
DEV
Screenshot 2025-07-14 at 17 22 12
